utils/env_preferences/tests/test.rs

```diff
@@ -0,0 +1,153 @@
+        }
+    }
+
+    #[test]
+    fn test_converting_locales() {
+        let locale_res: std::collections::HashMap<LocaleCategory, String> = get_locales().unwrap();
+        for locale in locale_res.into_values() {
+            let parts: Vec<&str> = locale.split('.').collect();
+
+            // Skipping "C" and those ending with "UTF-8", as they cannot be converted
+            // into the locale
+            if !parts.iter().any(|&part| part == "C")
+                && (parts.len() > 1 && parts[parts.len() - 1] != "UTF-8")
+            {
+                let mut locale_converted: Locale = locale.parse().unwrap();
+                locale_converted.extensions.unicode.clear();
+                assert_eq!(locale_converted, locale.parse().unwrap());
+            }
+        }
+    }
+
+    // This test contains unsafe code, the idea is to manually set a locale for `LC_TIME`,
+    // compare the result from `get_locales` and `get_system_calendar` they must be equal
+    #[test]
+    fn test_calendar() {
+        // Using "C" locale since it is the default, using any other locale like `en_IN` or `en_US`
+        // may work on some system and may not others depending on the availability
+        let test_calendar_locale = "C";
+        let locale_cstr =
+            std::ffi::CString::new(test_calendar_locale).expect("CString::new failed");
+
+        // SAFETY: This call is safe because any subsequent call to `setlocale` we pass a `NULL` locale
+        // to retrieve locale which does not sets the locale. The test locale `locale_cstr` is a CString
+        // nul terminated string for which we have the ownership
+        let tr = unsafe { setlocale(libc::LC_TIME, locale_cstr.as_ptr()) };
+
+        if tr.is_null() {
+            panic!("{:?}", RetrievalError::NullPointer);
+        }
+
+        let calendar_locale = get_system_calendars().unwrap();
+        assert_eq!(test_calendar_locale.to_string(), calendar_locale);
+    }
+}
+
+#[cfg(target_os = "macos")]
+#[cfg(test)]
+mod macos_test {
+    use env_preferences::{get_locales, get_system_calendars, get_system_timezone};
+    use icu_locale::Locale;
+
+    #[test]
+    fn test_get_locales() {
+        let locales_res = get_locales();
+        match locales_res {
+            Ok(locales) => {
+                assert!(!locales.is_empty(), "Unable to retrieve locales for Apple")
+            }
+            Err(e) => {
+                panic!("{:?}", e)
+            }
+        }
+    }
+
+    #[test]
+    fn test_converting_locales() {
+        let locales = get_locales().unwrap();
+        for locale in locales {
+            let _loc: Locale = locale.parse().unwrap();
+        }
+    }
+
+    #[test]
+    fn test_calendar() {
+        let calendar_res = get_system_calendars().unwrap();
+        for calendar in calendar_res {
+            assert!(!calendar.0.is_empty(), "Couldn't retreive calendar locale");
+            assert!(!calendar.1.is_empty(), "Couldn't retreive calendar");
+        }
```

Comment:
Nit: Testing nonempty is not sufficient. Please actually read the strings. Assert _something_ about their shape. Maybe have a list of valid values? At the very least check that they are ASCII-alphabetic, for example.

Reply:
The issue with a list is, we are not sure about the people who are contributing to the icu4x, they could have some different locale and it would display failing test. 

Check for ascii can be done